### DISCUSSION TITLE
DBeaver | 23.1.2 | How to: Remove Default Connection Types

### DISCUSSION THREAD

**Discussion by Yuuino** [dbeaver/dbeaver#22332] [Q&A]
I need to remove the default connection types
Development, Test, Production

I've created a custom join type, I need to remove the default values and leave 1 type
After that, transfer the config to several devices

Is it possible?

**Comment by LonwoLonwo** [comment]
Hello @Yuuino 

Nope, this is not possible. These types are the default types for DBeaver.
Do you have any specific reason to delete them at all?

**Comment by Yuuino** [comment]
Hello,
We have a database with a lot of active users who are using the editor incorrectly
You need to lock it for incorrect changes
Would it be a solution to assign the attribute to all connection types? 

modifyPermission="PERMISSION_EDIT_METADATA"
AppData\Roaming\DBeverData\workspace6\.metadata\.config\connection-types.xml

**Comment by Yuuino** [comment]
There are tables that indicate the accepted positions in the document
Support staff displays several cells in sql without explicit keys
And through the editor they change the data
Which entails adjusting all matching values in the entire table rather than at a specific local

**Comment by Yuuino** [comment]
Preventing the problem by explaining it will not completely eliminate the problem.
Therefore, a config is needed

**Comment by serjiokov** [comment]
@Yuuino Have you considered restricting permissions of database modification on server side?

**Comment by Yuuino** [comment]
Do you suggest restricting update/deletion ?
No matter how bad the users are, they need update permissions, in this case the question arises in configs Database tools

**Comment by Yuuino** [comment]
@LonwoLonwo, Добрый день :) Если гугл переводчик переводит криво, могу на Русском)

**Comment by LonwoLonwo** [comment]
@Yuuino No need, thanks.

The problem here is with these default connection types, which are hardcoded in our code.

Also, it looks like you need some permissions model. Which we have in our Team Edition product https://dbeaver.com/dbeaver-team-edition/.
Or in our cloud product: https://github.com/dbeaver/cloudbeaver

But no permission model in CE. Maybe closer is this: https://github.com/dbeaver/dbeaver/pull/21480#issuecomment-1755691891
